operator/internal/deploy/chart_manager.go

```diff
@@ -0,0 +1,222 @@
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/kyma-project/lifecycle-manager/operator/api/v1alpha1"
+	"github.com/kyma-project/lifecycle-manager/operator/pkg/remote"
+	modulelib "github.com/kyma-project/module-manager/operator/pkg/manifest"
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	k8syaml "sigs.k8s.io/yaml"
+)
+
+var ErrExpectedExactlyOneSKRConfig = errors.New("expected exactly one SKR config")
+
+type SKRWebhookChartManager interface {
+	InstallWebhookChart(ctx context.Context, kyma *v1alpha1.Kyma,
+		remoteClientCache *remote.ClientCache, kcpClient client.Client) (bool, error)
+	RemoveWebhookChart(ctx context.Context, kyma *v1alpha1.Kyma,
+		syncCtx *remote.KymaSynchronizationContext) error
+}
+
+type DisabledSKRWebhookChartManager struct{}
+
+//nolint:ireturn
+func ResolveSKRWebhookChartManager(isWatcherEnabled bool, skrConfigs ...*SkrChartConfig,
+) (SKRWebhookChartManager, error) {
+	if isWatcherEnabled && len(skrConfigs) != 1 {
+		return nil, ErrExpectedExactlyOneSKRConfig
+	}
+	if !isWatcherEnabled {
+		return &DisabledSKRWebhookChartManager{}, nil
+	}
+
+	return NewEnabledSKRWebhookChartManager(skrConfigs[0]), nil
+}
+
+func (m *DisabledSKRWebhookChartManager) InstallWebhookChart(_ context.Context, _ *v1alpha1.Kyma,
+	_ *remote.ClientCache, _ client.Client,
+) (bool, error) {
+	return false, nil
+}
+
+func (m *DisabledSKRWebhookChartManager) RemoveWebhookChart(_ context.Context, _ *v1alpha1.Kyma,
+	_ *remote.KymaSynchronizationContext,
+) error {
+	return nil
+}
+
+type EnabledSKRWebhookChartManager struct {
+	config  *SkrChartConfig
+	kcpAddr string
+}
+
+type SkrChartConfig struct {
+	// WebhookChartPath represents the path of the webhook chart
+	// to be installed on SKR clusters upon reconciling kyma CRs.
+	WebhookChartPath             string
+	SkrWebhookMemoryLimits       string
+	SkrWebhookCPULimits          string
+	EnableWebhookPreInstallCheck bool
+}
+
+func NewEnabledSKRWebhookChartManager(config *SkrChartConfig) *EnabledSKRWebhookChartManager {
+	return &EnabledSKRWebhookChartManager{
+		config: config,
+	}
+}
+
+func (m *EnabledSKRWebhookChartManager) InstallWebhookChart(ctx context.Context, kyma *v1alpha1.Kyma,
+	remoteClientCache *remote.ClientCache, kcpClient client.Client,
+) (bool, error) {
+	skrClient, err := remote.NewRemoteClient(ctx, kcpClient, client.ObjectKeyFromObject(kyma),
+		kyma.Spec.Sync.Strategy, remoteClientCache)
+	if err != nil {
+		return true, err
+	}
+	skrCfg, err := remote.GetRemoteRestConfig(ctx, kcpClient, client.ObjectKeyFromObject(kyma),
+		kyma.Spec.Sync.Strategy)
+	if err != nil {
+		return true, err
+	}
+	argsVals, err := m.generateHelmChartArgs(ctx, kcpClient)
+	if err != nil {
+		return true, err
+	}
+	// TODO(khlifi411): make sure that validating-webhook-config resource is in sync with the secret configuration
```

Comment:
Resolve TODO

Reply:
As discussed. This is a bigger refactoring which will be done [here](https://github.com/kyma-project/lifecycle-manager/issues/301)

Reply:
Why is this part of bigger refactoring? @adityabhatia @ruanxin